client/state/login/oauth2/reducer.js

```diff
@@ -71,11 +73,29 @@ export const initialClientsData = {
 	},
 };
 
+const getClientIdFromSignupUrl = ( signupUrl ) => {
+	const urlParts = signupUrl.split( '?' );
+	const queryString = urlParts[ 1 ];
```

Comment:
There's a lib we use called `url` to do this elsewhere. might be a safer approach...